op-supervisor/supervisor/backend/db/heads/types.go

```diff
@@ -0,0 +1,81 @@
+package heads
+
+import (
+	"encoding/json"
+
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db/entrydb"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// ChainHeads provides the serialization format for the current chain heads.
+// The values here could be block numbers or just the index of entries in the log db.
+// If they're log db entries, we can't detect if things changed because of a reorg though (if the logdb write succeeded and head update failed).
+// So we probably need to store actual block IDs here... but then we don't have the block hash for every block in the log db.
+// Only jumping the head forward on checkpoint blocks doesn't work though...
+type ChainHeads struct {
```

Comment:
I'm fine merging this as-is, but we'll need to iterate on the approach here, as you mentioned in the PR description